### ISSUE TITLE
Zerocash paper: DAP Receive needs coins as input

### ISSUE THREAD

**Issue by defuse** [Electric-Coin-Company/zcash#722]
Minor error in the Zerocash paper:

In section 3.2 it's describing DAP schemes at a high level, and at this stage txMint and txPour do not contain the coins themselves. So the paper hasn't (yet) mentioned the coins (boldface c's) being stored on the ledger. The mistake is that Receive somehow outputs a list of unspent coins given only the ledger and address key pair as input. That's not necessarily possible given what's previously described in that section.

I think Receive actually needs a list of coins as part of its input, and it will return the subset of those that are unspent on the ledger. At least at this stage of the description.

**Comment by daira** [comment]
Yeah, this is inconsistent. Receive as written doesn't make sense unless the particular DAP scheme supports sending coins via the ledger. If they are sent out-of-band, however, you still need a way to verify that they are valid.

**Comment by daira** [comment]
The cleanest way I can think of to fix this would be to split Receive into ScanForCoins and VerifyCoin, with the former calling the latter for each purported coin it finds.

**Comment by daira** [comment]
I will contact the scientists about this.

**Comment by defuse** [comment]
@matthewdgreen: This ticket may interest you.
